engines/tot/resources.cpp

```diff
@@ -0,0 +1,395 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "tot/tot.h"
+#include "tot/util.h"
+
+namespace Tot {
+
+void TotEngine::readBitmap(int32 bitmapPosition, byte *buf, uint bitmapSize, uint error) {
+	Common::File bitmapFile;
+	if (!bitmapFile.open("BITMAPS.DAT")) {
+		showError(error);
+	}
+	bitmapFile.seek(bitmapPosition);
+	bitmapFile.read(buf, bitmapSize);
+
+	bitmapFile.close();
+}
+
+void TotEngine::loadCharAnimation() {
+	Common::File characterFile;
+	_cpCounter = _cpCounter2;
+	if (!characterFile.open("PERSONAJ.SPT"))
+		showError(265);
+
+	_mainCharFrameSize = characterFile.readUint16LE();
+
+	verifyCopyProtection();
+
+	for (int i = 0; i <= 3; i++)
+		for (int j = 0; j < kWalkFrameCount; j++) {
+			_mainCharAnimation.bitmap[i][j] = (byte *)malloc(_mainCharFrameSize);
+			characterFile.read(_mainCharAnimation.bitmap[i][j], _mainCharFrameSize);
+		}
+	for (int i = 0; i < 4; i++)
+		for (int j = kWalkFrameCount; j < (kWalkFrameCount + 10 * 3); j++) {
+			_mainCharAnimation.bitmap[i][j] = (byte *)malloc(_mainCharFrameSize);
+			characterFile.read(_mainCharAnimation.bitmap[i][j], _mainCharFrameSize);
+		}
+	characterFile.close();
+}
+
+void TotEngine::readTextFile() {
+	if (!_verbFile.open("CONVERSA.TXT")) {
+		showError(313);
+	}
+}
+
+TextEntry TotEngine::readTextRegister(uint numRegister) {
+	_verbFile.seek(numRegister * kVerbRegSize);
+	return readTextRegister();
+}
+
+TextEntry TotEngine::readTextRegister() {
+	TextEntry regmht;
+	// Since the text is encrypted it's safer to save the size as reported by
+	// the pascal string, because the encrypter character might be the termination
+    // character
+	byte size = _verbFile.readByte();
+	_verbFile.seek(-1, SEEK_CUR);
+	regmht.text = _verbFile.readPascalString(false);
+	_verbFile.skip(255 - size);
+	regmht.continued = _verbFile.readByte();
+	regmht.response = _verbFile.readUint16LE();
+	regmht.pointer = _verbFile.readSint32LE();
+	return regmht;
+}
+
+void TotEngine::readConversationFile() {
+	Common::File conversationFile;
+	if (!conversationFile.open(Common::Path(Common::String("CONVERSA.TRE")))) {
+		showError(314);
+	}
+	int64 fileSize = conversationFile.size();
+	byte *buf = (byte *)malloc(fileSize);
+	conversationFile.read(buf, fileSize);
+
+	_conversationData = new Common::MemorySeekableReadWriteStream(buf, fileSize, DisposeAfterUse::NO);
+	conversationFile.close();
+}
+
+void saveDoorMetadata(DoorRegistry doors, Common::SeekableWriteStream *screenDataStream) {
```

Comment:
```suggestion
void saveDoorMetadata(DoorRegistry doors, Common::SeekableWriteStream *out) {
```

for brevity, and the same below